components/support/jwcrypto/src/direct.rs

```diff
@@ -0,0 +1,159 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+// Support for "Direct Encryption with a Shared Symmetric Key"
+use crate::{
+    aes,
+    error::{JwCryptoError, Result},
+    Algorithm, CompactJwe, EncryptionAlgorithm, JweHeader, Jwk, JwkKeyParameters,
+};
+use rc_crypto::rand;
+
+impl Jwk {
+    /// Create a new random key suitable for `Direct` symmetric encryption.
+    /// Consumers can store this for later use, probably via serde serialization.
+    pub fn new_symmetric_key(kid: Option<String>) -> Result<Self> {
+        // We only support AES256 which as a 32byte key.
+        let mut bytes: Vec<u8> = vec![0; 32];
+        rand::fill(&mut bytes)?;
+        Ok(Jwk {
+            kid,
+            key_parameters: JwkKeyParameters::Direct(bytes),
+        })
+    }
+
+    // Create a new Jwk given the raw bytes of a `Direct` key. We generally
+    // prefer consumers to use serde with the entire key and the only use-case
+    // is currently tests, so let's limit this to tests for now...
+    #[cfg(test)]
+    pub fn new_symmetric_from_bytes(kid: Option<String>, key: &[u8]) -> Self {
+        Jwk {
+            kid,
+            key_parameters: JwkKeyParameters::Direct(key.to_vec()),
+        }
+    }
+}
+
+pub(crate) fn encrypt_to_jwe_direct(
```

Comment:
naming nit: `direct` is kind of implied by the name of the module and may not be needed in the name of the function here (although I don't feel strongly about it either way).